### ISSUE TITLE
Problem with abstract models deserialization

### ISSUE THREAD

**Issue by gaperton** [Volicon/backbone.nestedTypes#79]
The way it's supposed to be done now is overriding 'create' in attribute metatype. Which, in fact, breaks 'create' method's semantic, and won't work in all situations.

The right way to do it is, possibly, to override 'parse'. In this case it will work in more consistent way.

This approach require no codebase modification, but I will leave this ticket here for the case I will invent something more fancy.

**Comment by kkilictepe** [comment]
I`ve also spent some time on this issue and deserialization of abstract model collections as well.
I think overriding 'parse' is the only option but please let us know if you can find a better way. 
I`d be very ashamed if you saw my solution at the moment.

**Comment by gaperton** [comment]
Well, that's mine current solution. I've got abstract Widget model, which is a base class for other Widgets. I added factory method Widgets.fromJSON to create proper subcllass, and then it looks like this:

```javascript
var Cell = Model.extend({
        defaults : {
            x : 0,
            y : 0,
            w : 1,
            h : 1,
            widget : Widget
        },

        parse : function( resp ){
            resp.widget = Widget.fromJSON( resp.widget );
            return resp;
        },
        ...
});
```

The trick is that you have to assign Widget.fromJSON factory from some separate module, which will include all the subclasses, or you will got circular dependency.

It's not so bad.

**Comment by gaperton** [comment]
Ok, it seems that attribute's 'parse' option is the best way.

```javascript
var Cell = Model.extend({
        defaults : {
            x : 0,
            y : 0,
            w : 1,
            h : 1,
            widget : Widget.has.parse( function( resp ){
                return Widget.fromJSON( resp );
            }),
        ...
});
```

Just implemented in 1.0.0 ('develop') highly efficient 'parse' option compiler step, in the way that it won't slow down regular code. Tested on volicon sources. Works like a charm.
